### PR TITLE
Migrates `prefect-redis` to the core repository

### PR DESCRIPTION
**Pull request by desertaxle**
Migrates the code for `prefect-redis` from its current repository into the core library with the rest of the integrations. This PR doesn't make any code changes, but subsequent PRs will migrate Redis functionality in the core library to `prefect-redis` and add a `RedisRecordStore` implementation.

Related to https://github.com/PrefectHQ/prefect/issues/14914

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: .github/workflows/integration-package-tests.yaml [hunk 1]
```diff
@@ -90,6 +90,15 @@ jobs:
         if: matrix.package == 'prefect-docker'
         run : prefect dev build-image
 
+      - name: Start redis
+        if: matrix.package == 'prefect-redis'
+        run: >
+          docker run
+          --name "redis"
+          --detach
+          --publish 6379:6379
```
Comment (zzstoatzz) [inline .github/workflows/integration-package-tests.yaml:99]: TIL `-p` is `publish` and not `port`

File: src/integrations/prefect-redis/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+## Welcome!
+
+Prefect integrations for working with Redis
+
+## Getting Started
+
+### Python setup
+
+Requires an installation of Python 3.7+.
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:27]:
```suggestion
Requires an installation of Python 3.9+.
```

File: src/integrations/prefect-redis/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+### Installation
+
+Install `prefect-redis` with `pip`:
+
+```bash
+pip install prefect-redis
+```
+
+Then, register to [view the block](https://orion-docs.prefect.io/ui/blocks/) on Prefect Cloud:
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:41]:
```suggestion
Then, register to view the block on Prefect Cloud:
```

File: src/integrations/prefect-redis/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+## Getting Started
+
+### Python setup
+
+Requires an installation of Python 3.7+.
+
+We recommend using a Python virtual environment manager such as pipenv, conda or virtualenv.
+
+These tasks are designed to work with Prefect 2.0. For more information about how to use Prefect, please refer to the [Prefect documentation](https://orion-docs.prefect.io/).
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:31]:
```suggestion
These tasks are designed to work with Prefect 2.0. For more information about how to use Prefect, please refer to the [Prefect documentation](https://docs.prefect.io/).
```

File: src/integrations/prefect-redis/README.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+pip install prefect-redis
+```
+
+Then, register to [view the block](https://orion-docs.prefect.io/ui/blocks/) on Prefect Cloud:
+
+```bash
+prefect block register -m prefect_redis.credentials
+```
+
+Note, to use the `load` method on Blocks, you must already have a block document [saved through code](https://orion-docs.prefect.io/concepts/blocks/#saving-blocks) or [saved through the UI](https://orion-docs.prefect.io/ui/blocks/).
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:47]:
```suggestion
Note, to use the `load` method on Blocks, you must already have a block document [saved through code](https://docs.prefect.io/concepts/blocks/#saving-blocks) or [saved through the UI](https://docs.prefect.io/ui/blocks/).
```

File: src/integrations/prefect-redis/README.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+example_flow()
+```
+
+## Resources
+
+If you encounter any bugs while using `prefect-redis`, feel free to open an issue in the [prefect-redis](https://github.com/C4IROcean/prefect-redis) repository.
+
+If you have any questions or issues while using `prefect-redis`, you can find help in either the [Prefect Discourse forum](https://discourse.prefect.io/) or the [Prefect Slack community](https://prefect.io/slack).
+
+Feel free to ⭐️ or watch [`prefect-redis`](https://github.com/C4IROcean/prefect-redis) for updates too!
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:83]:
```suggestion
```

File: src/integrations/prefect-redis/README.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+```bash
+git clone https://github.com/C4IROcean/prefect-redis.git
+
+cd prefect-redis/
+
+pip install -e ".[dev]"
+
+# Install linting pre-commit hooks
+pre-commit install
+```
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:98]: i think this whole section should be updated or removed

**Review by zzstoatzz — COMMENTED** [review]
a couple comments on the readme / links

File: src/integrations/prefect-redis/README.md (hunk 7 above)
Comment (desertaxle) [inline src/integrations/prefect-redis/README.md:98]:
```suggestion
```

Comment (desertaxle) [reply src/integrations/prefect-redis/README.md]: Agreed, I removed it for now.

File: src/integrations/prefect-redis/README.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+
+<p align="center">
+    <a href="https://pypi.python.org/pypi/prefect-redis/" alt="PyPI version">
+        <img alt="PyPI" src="https://img.shields.io/pypi/v/prefect-redis?color=0052FF&labelColor=090422"></a>
+    <a href="https://github.com/C4IROcean/prefect-redis/" alt="Stars">
+        <img src="https://img.shields.io/github/stars/C4IROcean/prefect-redis?color=0052FF&labelColor=090422" /></a>
+    <a href="https://pepy.tech/badge/prefect-redis/" alt="Downloads">
+        <img src="https://img.shields.io/pypi/dm/prefect-redis?color=0052FF&labelColor=090422" /></a>
+    <a href="https://github.com/C4IROcean/prefect-redis/pulse" alt="Activity">
+        <img src="https://img.shields.io/github/commit-activity/m/C4IROcean/prefect-redis?color=0052FF&labelColor=090422" /></a>
```
Comment (zzstoatzz) [inline src/integrations/prefect-redis/README.md:11]: i think we also wanna nix these right?

**Review by zzstoatzz — APPROVED** [review]
lgtm!

**Pull request merged**